### PR TITLE
feat(Amazon S3): Make direct href links public and organize media fil…

### PR DESCRIPTION
**Pull request by ioanpocol**
…e by date

### CODE REVIEW

**Comment by plamut** [comment]
Do we test these methods?
The change itself seems fine.

File: superdesk/storage/amazon/amazon_media_storage.py [hunk 1]
```diff
@@ -59,11 +60,20 @@ def url_for_media(self, media_id):
...
+
+        return '%s/%s' % (time.strftime('%Y%m%d'), media_id)
+
     def read_from_config(self):
         self.region = self.app.config.get('AMAZON_REGION', 'us-east-1') or 'us-east-1'
         username = self.app.config['AMAZON_ACCESS_KEY_ID']
         api_key = self.app.config['AMAZON_SECRET_ACCESS_KEY']
         self.container_name = self.app.config['AMAZON_CONTAINER_NAME']
+        self.kwargs = {}
+        if self.app.config.get('AMAZON_SERVE_DIRECT_LINKS', False):
```
Comment (petrjasek) [inline superdesk/storage/amazon/amazon_media_storage.py]: maybe this should be `PUBLIC_LINKS` not `DIRECT_LINKS` it would be more obvious imo

**Comment by petrjasek** [comment]
:+1: some tests would be great (as long as it would talk to real s3)

File: superdesk/storage/amazon/amazon_media_storage.py (hunk 1 above)
Comment (ioanpocol) [inline superdesk/storage/amazon/amazon_media_storage.py]: Public links are just first version but the intention is to have direct links but not public.

**Comment by ioanpocol** [comment]
I think that for behave tests we have to make the amazon account credentials visible on github, so because of this there are no tests for this.

**Comment by petrjasek** [comment]
so I suggest define tests talking to s3 using environment vars - in case it's not set do nothing - and we can set those credentials on bamboo.. but that's task of its own, so wouldn't block this meanwhile

**Comment by ioanpocol** [comment]
Added a new task for behave tests: SDPA-41

**Comment by petrjasek** [comment]
@ioanpocol are you going to merge this to pa-poc branch too?

**Comment by ioanpocol** [comment]
I will update the superdek-core version on requirements.txt on both master and pa-poc branches.

**Pull request merged**